worker_specific_task_queues/worker_specific_activities.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+require 'digest'
+require 'logger'
+require 'net/http'
+require 'tempfile'
+require 'temporalio/activity'
+require 'uri'
+
+module WorkerSpecificTaskQueues
+  module Activities
+    class DownloadFileActivity < Temporalio::Activity::Definition
+      def execute(url)
+        # Simulate slow activity
+        sleep(3)
+
+        # Download file using block syntax for automatic cleanup
+        temp_path = nil
+        Tempfile.open do |file|
```

Comment:
I am concerned using `.open` will unlink (i.e. delete) the file when the object is GC'd

Reply:
Good catch. Guess I've been getting lucky with GC not kicking in during the life of the workflow.